src/client.rs

```diff
@@ -111,28 +138,124 @@ mod test {
         let child = thread::spawn(move || {
             let (mut connection, _)  = test_server.accept().unwrap();
 
+            // First proposal should be fine, no errors.
+            scoped_debug!("Should get preamble and proposal. Responds Success");
```

Comment:
I'm not a huge fan of debug logs in tests, a lot of times the log message would be better as an assertion that whatever the condition is holds.  I'm not too hard and fast about this, though.